### PR TITLE
better python and form generator

### PR DESCRIPTION
**Pull request by giulianorosella**
better formulas generators in python

### CODE REVIEW

File: src/python/formulas_generator.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
+from middle_step import binary_comb_prov9, binary_comb_z3
+
+from vars import atoms_for_prover, atoms_for_z3, con_numbers, var_numbers
+
+from typing import Any, List
+
+
+# ignore the following commentes function
```
Comment (zazangra) [inline src/python/formulas_generator.py]: Remove commented code

File: src/python/formulas_generator.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                    formulas_prov9(index[h][0]), formulas_prov9(index[h][1])
+                )
+    return form
+
+
+# function that takes as an input an integer n and generates all the formulas - in the language for prover9 - with n
+# connectives starting from our initial propositional variables atoms_for_prover
+
+
+print(len(formulas_prov9(con_numbers)))
```
Comment (zazangra) [inline src/python/formulas_generator.py]: debug using debugger, don't print (and remove please)

File: src/python/middle_step.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
+from z3 import Implies, And, Or  # type: ignore
+
+from typing import Any, List
+
+
+def binary_comb_z3(list1: List[Any], list2: List[Any]) -> List[Any]:
```
Comment (zazangra) [inline src/python/middle_step.py]: Can we remove any and be more explicit?

File: src/python/vars.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+)  # the list of the range of pur propositional variable, i.e. all the uppercase letters in the alphabet
+
+var_numbers = 5  # here the only thing we can change, ie- the number of variables we want to have
+
+atoms_for_prover: List[str] = ["0"] + range_prop_vars[0:var_numbers]  # type: ignore # the list of our atomic formulas that can be the inputs for prover/mace4, i.e. propositional variables plus the bottom element which is the int 0
+
+bool_vars = []
+
+for i in range(var_numbers):
+    lettera = range_prop_vars[i]
```
Comment (zazangra) [inline src/python/vars.py]: don't mix language, stick to english

File: src/python/vars.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    ascii_uppercase
+)  # the list of the range of pur propositional variable, i.e. all the uppercase letters in the alphabet
+
+var_numbers = 5  # here the only thing we can change, ie- the number of variables we want to have
+
+atoms_for_prover: List[str] = ["0"] + range_prop_vars[0:var_numbers]  # type: ignore # the list of our atomic formulas that can be the inputs for prover/mace4, i.e. propositional variables plus the bottom element which is the int 0
+
+bool_vars = []
+
+for i in range(var_numbers):
```
Comment (zazangra) [inline src/python/vars.py]: make this a func and call it here

File: src/python/vars.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
+# this file contains all the initial imput parameters that we may want to change for our investigation
```
Comment (zazangra) [inline src/python/vars.py]: move this file inside a folder, ex: python/vars/vars.py

File: src/python/middle_step.py [hunk 7]
```diff
@@ -0,0 +1,67 @@
+from z3 import Implies, And, Or  # type: ignore
```
Comment (zazangra) [inline src/python/middle_step.py]: move this file inside a folder /utils/comb.py

File: src/python/vars.py (hunk 5 above)
Comment (zazangra) [reply src/python/vars.py]: we can move this function inside utils in another file , ex /utils/formulas.py

**Review by zazangra — CHANGES_REQUESTED** [review]
resolve comment and we're good to go

File: src/python/formulas_generator.py (hunk 1 above)
Comment (giulianorosella) [reply src/python/formulas_generator.py]: done

File: src/python/formulas_generator.py (hunk 2 above)
Comment (giulianorosella) [reply src/python/formulas_generator.py]: done

File: src/python/middle_step.py (hunk 3 above)
Comment (giulianorosella) [reply src/python/middle_step.py]: honestly, I haven't found what type those things are

File: src/python/vars.py (hunk 4 above)
Comment (giulianorosella) [reply src/python/vars.py]: sure

File: src/python/vars.py (hunk 5 above)
Comment (giulianorosella) [reply src/python/vars.py]: done

**Review by zazangra — APPROVED** [review]
Great work!! :punch: :boom: :fire:

**Comment by zazangra** [comment]
Hey @giulianorosella  seems we have conflicts with main. Will resolve and merge

**Pull request merged**